hazelcast/src/main/java/com/hazelcast/map/impl/SizeEstimator.java

```diff
@@ -23,11 +23,17 @@
  */
 public interface SizeEstimator<T> {
 
+    /**
+     * Assume compressed object reference is used.
```

Comment:
This is a bad assumption. References can be 32 or 64 bit depending on JVM being 32 or 64 and OOP compression.